walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,111 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Collection;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
 
+/** Find available meeting time slots */
 public final class FindMeetingQuery {
+  /** Get events of requested attendees */
+  private List<Event> getRelevantEvents(Collection<Event> events, Collection<String> requestedAttendee){
+    List<Event> relevantEvents = new ArrayList<>();
+    Iterator<Event> iter = events.iterator();
+    Event currEvent;
+    Set<String> meetingAttendee = new HashSet<>();
+    while(iter.hasNext()){
+      currEvent = iter.next();
+      meetingAttendee.addAll(currEvent.getAttendees());
+      meetingAttendee.retainAll(requestedAttendee);
+      if(meetingAttendee.size() != 0){
+        relevantEvents.add(currEvent);
+      }
+      meetingAttendee.clear();
+    }
+
+    return relevantEvents;
+  }
+
+  public static final Comparator<Event> SORT_EVENT_BY_START = new Comparator<Event>() {
+    @Override
+    public int compare(Event a, Event b) {
+      return Long.compare(a.getWhen().start(), b.getWhen().start());
+    }
+  };
+
+  /** check if time from start to end is at least length of duration */
+  private boolean checkTimeRange (int start, int end, long duration) {
+    if (((long)(end - start)) < duration) {
+      return false;
+    }
+    return true;
+  }
+
+  /** Find available timeranges given requested attendees */
+  public Collection<TimeRange> queryHelper(Collection<Event> events, 
+  Collection<String> attendees, long duration) {
+    List<Event> relevantEvents = getRelevantEvents(events, attendees);
+    relevantEvents.sort(SORT_EVENT_BY_START);
+
+    int start = TimeRange.START_OF_DAY;
+    int end = TimeRange.START_OF_DAY;
+    int currStart;
+    TimeRange prevTimeRange = TimeRange.fromStartDuration(0, 0);
+    TimeRange currTimeRange;
+    Event prevEvent = null;
+    Collection<TimeRange> possibleTime = new ArrayList<>();
+    
+    for(Event currEvent: relevantEvents) {
+      currTimeRange = currEvent.getWhen();
+      currStart = currTimeRange.start();
+      if (prevEvent == null && currStart != TimeRange.START_OF_DAY) {
```

Comment:
I took out this if check now since I previously was thinking about preventing a null exception error (which in this case doesn't happen in java:)) for prevent and trying to use this case as a special case for the first check